library/alloc/src/vec/mod.rs

```diff
@@ -1074,8 +1074,9 @@ impl<T, A: Allocator> Vec<T, A> {
 
     /// Converts the vector into [`Box<[T]>`][owned slice].
     ///
-    /// If the vector has excess capacity, its items will be moved into a
-    /// newly-allocated buffer with exactly the right capacity.
+    /// If the vector has excess capacity, it will shrink down as close as
+    /// possible to the length but the allocator may still inform the vector
+    /// that there is space for a few more elements.
```

Comment:
I don't think the "few more elements" part is relevant because it'll still [fit](https://doc.rust-lang.org/nightly/std/alloc/trait.Allocator.html#memory-fitting) the Box layout.
I think it should just link to `Allocator::shrink` which can explain the details.

Reply:
Yes, I was just looking at that bit about layout fitting. It might be good to avoid linking to low-level documentation for a higher-level method though (without a clear, high-level explanation). Another option is to just link to shrink_to_fit, and maybe to improve the shrink_to_fit documentation and have *that* link to Allocator::shrink.

Reply:
Made the changes according to what I said above.